key/src/main/java/software/amazon/kms/key/UpdateHandler.java

```diff
@@ -58,9 +58,12 @@ public UpdateHandler(final ClientBuilder clientBuilder,
                 .updateKeyDescription(proxy, proxyClient, previousModel, model, callbackContext))
             .then(progress -> keyHandlerHelper
                 .updateKeyPolicy(proxy, proxyClient, previousModel, model, callbackContext))
-            .then(progress -> softFailAccessDenied(() -> keyHandlerHelper
-                .updateKeyTags(proxy, proxyClient, model, tags, callbackContext),
-                model, callbackContext))
+            .then(progress -> {
+                if (!tags.equals(request.getPreviousResourceTags())) {
+                    keyHandlerHelper.updateKeyTags(proxy, proxyClient, model, tags, callbackContext);
```

Comment:
You should return this. Otherwise I believe this will end up swallowing all errors thrown by the tagging APIs.

Reply:
+1